### ISSUE TITLE
CDR code: Run time error

### ISSUE THREAD

**Issue by jppelteret** [dealii/code-gallery#76]
@drwells Does the CDR example still run for you? I get an error message when I try to run it (Trilinos 12.18.1)
```
<PATH>/cdr/build $ cp ../parameters.prm .
<PATH>/cdr/build $ make run
[100%] Run with Debug configuration
Number of degrees of freedom: 2720

*********************************************************
ML failed to compute the multigrid preconditioner. The
most common problem is an incorrect  data type in ML's
parameter list (e.g. 'int' instead of 'bool').

Note: List.set("ML print initial list",X) might help
figure out the bad one on pid X.

The following exception was thrown:
Error!  An attempt was made to access parameter "initialize random seed" of type "bool"
in the parameter (sub)list "SA default values"
using the incorrect type "bool"!

Throw number = 1

*********************************************************

ML::ERROR:: -1, /<PATH>/trilinos-12.18.1-l7t2hzrmefrv5zd4crr4a2ukhx2jgtuw/spack-src/packages/ml/src/Utils/ml_MultiLevelPreconditioner.cpp, line 2566
ML::ERROR::
*********************************************************
ML failed to compute the multigrid preconditioner. The
most common problem is an incorrect  data type in ML's
parameter list (e.g. 'int' instead of 'bool').

Note: List.set("ML print initial list",X) might help
figure out the bad one on pid X.

The following exception was thrown:
Error!  An attempt was made to access parameter "initialize random seed" of type "bool"
in the parameter (sub)list "SA default values"
using the incorrect type "bool"!

Throw number = 2

*********************************************************
```

**Comment by drwells** [comment]
Hm, interesting - I actually did run it when getting ready for the release to check #73 et al. I'll take a look.

**Comment by drwells** [comment]
The example still runs just fine for me. I only use default options for the AMG preconditioner so if there is an issue it is probably upstream and has since been fixed (I'm using Trilinos 13.2)
